### PR TITLE
Resources and SitePage refactoring

### PR DESCRIPTION
**Pull request by akench**
Refactored site pages to open in a new SitePageActivity, and the activity will load the correct fragment, depending on the site page.
Site pages without a custom fragment will be loaded as a web view
Also resources are showing and downloading,
adding icons to resources and removing the SwipeRefreshLayout will be in another pull request

### CODE REVIEW

File: app/src/main/java/com/sakaimobile/development/sakaiclient20/repositories/ResourceRepository.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    /**
+     * refresh the site resources, sets the siteIDs and then persists them
+     * @param siteId siteId of resources to request
+     * @return whether or not the operation was successful
+     */
+    public Single<List<Resource>> refreshSiteResources(String siteId) {
+        return resourcesService
+                .getSiteResources(siteId)
+                .map(ResourcesResponse::getResources)
+                .map(resources -> setSiteIdOfResources(siteId, resources))
```
Comment (SChakravorti21) [inline app/src/main/java/com/sakaimobile/development/sakaiclient20/repositories/ResourceRepository.java]: Oh I see, this is where the siteId is set on the resources. Then maybe change the comment in the deserializer and add a comment here to describe _why_ it's done this way, because the deserializer gives the impression that siteId is never set.

File: app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/SakaiApplication.java [hunk 2]
```diff
@@ -11,17 +12,20 @@
...
 
     @Inject DispatchingAndroidInjector<Activity> activityInjector;
 
+    @Inject DispatchingAndroidInjector<Fragment> supportFragmentInjector;
+
     @Override
     public void onCreate() {
         super.onCreate();
-        Fabric.with(this, new Crashlytics());
+//        Fabric.with(this, new Crashlytics());
```
Comment (SChakravorti21) [inline app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/SakaiApplication.java]: Don't comment this out, there's a proper way to disable Crashlytics in debug mode that's in the assignments branch and will get added when I make a separate PR.

File: app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/activities/MainActivity.java [hunk 3]
```diff
@@ -254,86 +225,13 @@ public void onAnnouncementSelected(Announcement announcement, Map<String, Course
...
         b.putSerializable(getString(R.string.siteid_to_course_map), (HashMap) siteIdToCourse);
-        b.putSerializable(getString(R.string.siteid_to_course_map), (HashMap) siteIdToCourse);
 
         SingleAnnouncementFragment fragment = new SingleAnnouncementFragment();
         fragment.setArguments(b);
 
         loadFragment(fragment, FRAGMENT_ADD, true, R.anim.grow_enter, R.anim.pop_exit);
     }
 
-    @Override
```
Comment (SChakravorti21) [inline app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/activities/MainActivity.java]: Awesome that all this stuff could be taken out of the activity!

File: app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/activities/SitePageActivity.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+    private void startWebViewFragment(String siteName, Course course) {
+
+        String url = null;
+        for(SitePage page : course.sitePages) {
+            if(page.title.equals(siteName))
+                url = page.url;
+        }
+
+        if(url != null) {
```
Comment (SChakravorti21) [inline app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/activities/SitePageActivity.java]: Is the URL ever null?

File: app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/activities/SitePageActivity.java [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+
+
+public class SitePageActivity extends AppCompatActivity {
+
+
+    protected Set<LiveData> beingObserved;
```
Comment (SChakravorti21) [inline app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/activities/SitePageActivity.java]: Doesn't seem like this is used for any ViewModels here in the activity

File: app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/adapters/AnnouncementsAdapter.java [hunk 6]
```diff
@@ -42,7 +43,7 @@
     // mapping siteIdToCourse, needed to get subject code and course title
     private HashMap<String, Course> siteIdToCourse;
```
Comment (SChakravorti21) [inline app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/adapters/AnnouncementsAdapter.java]: Can't this just be a String to String map for siteId -> course name if you only need the course title?

File: app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/fragments/CourseSitesFragment.java [hunk 7]
```diff
@@ -46,22 +55,47 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
...
+//        // TODO change the hackies
+//        MainActivity mainActivity = (MainActivity) getActivity();
+//        mainActivity.setActionBarTitle(courseToView.title);
 
 
         return inflated;
     }
+
+
+//    private void startAnnouncementsActivity() {
```
Comment (SChakravorti21) [inline app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/fragments/CourseSitesFragment.java]: If everything is commented out here can we take it out

File: app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/fragments/SiteResourcesFragment.java [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+     *
+     * @param resources list, must be in a proper traversal format, otherwise algorithm won't work
+     *                  vist, then go to all children, recursively
+     * @param start     start of the list to get children of
+     * @param end       end of the part of the list representing the descendants
+     * @return a list of nodes which are the children of the node at resources[start]
+     */
+    private List<TreeNode> getChildren(List<Resource> resources, int start, int end) {
+
+        List<TreeNode> children = new ArrayList<>();
```
Comment (SChakravorti21) [inline app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/fragments/SiteResourcesFragment.java]: Looks good

File: app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/fragments/SiteResourcesFragment.java [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+        List<TreeNode> children = node.getChildren();
+        while (children.size() > 0) {
+            TreeNode child = children.get(0);
+            resourcesTreeView.removeNode(child);
+        }
+    }
+
+
+
+//    private void setupToolbar(List<Resource> resources) {
```
Comment (SChakravorti21) [inline app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/fragments/SiteResourcesFragment.java]: same thing, if it's going to remain commented out then consider removing it.

File: app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/viewmodels/AnnouncementViewModel.java [hunk 10]
```diff
@@ -128,67 +152,18 @@ public void refreshSiteData(String siteId, int num) {
                         .subscribeOn(Schedulers.io())
                         .observeOn(AndroidSchedulers.mainThread())
                         .subscribe(
-                                () -> loadSiteAnnouncements(siteId),
+                                (resources) -> {
```
Comment (SChakravorti21) [inline app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/viewmodels/AnnouncementViewModel.java]: Should this be called announcements instead of resources

File: app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/activities/SitePageActivity.java (hunk 4 above)
Comment (akench) [reply app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/activities/SitePageActivity.java]: it shouldn't be, but its just a failsafe, should I remove the null check?

File: app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/adapters/AnnouncementsAdapter.java (hunk 6 above)
Comment (akench) [reply app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/adapters/AnnouncementsAdapter.java]: I also need the subject code of the course so that I can show the right icon
